src/validate_pyproject/api.py

```diff
@@ -185,16 +197,53 @@ def __getitem__(self, key: str) -> Callable[[str], Schema]:
         return self._registry.__getitem__
 
 
+def load_from_uri(tool_uri: str) -> Tuple[str, Any]:
+    tool_info = urllib.parse.urlparse(tool_uri)
+    if tool_info.netloc:
+        url = f"{tool_info.scheme}://{tool_info.netloc}{tool_info.path}"
+        with open_url(url) as f:
+            contents = json.load(f)
+    else:
+        with open(tool_info.path, "rb") as f:
+            contents = json.load(f)
+    return tool_info.fragment, contents
+
+
+class RemotePlugin:
+    def __init__(self, tool: str, fragment: str, schema: Schema):
+        self.id = schema["$id"]
+        self.tool = tool
+        self.schema = schema
+        self.help_text = f"{tool} <external>"
+        self.fragment = fragment
+
+
+if typing.TYPE_CHECKING:
+    from .plugins import PluginProtocol
+
+    _: PluginProtocol = typing.cast(RemotePlugin, None)
+
+
 class Validator:
+    _plugins: Sequence["PluginProtocol"]
+
     def __init__(
         self,
-        plugins: Union[Sequence["PluginWrapper"], AllPlugins] = ALL_PLUGINS,
+        plugins: Union[Sequence["PluginProtocol"], AllPlugins] = ALL_PLUGINS,
         format_validators: Mapping[str, FormatValidationFn] = FORMAT_FUNCTIONS,
         extra_validations: Sequence[ValidationFn] = EXTRA_VALIDATIONS,
+        load_tools: Sequence[str] = (),
     ):
         self._code_cache: Optional[str] = None
         self._cache: Optional[ValidationFn] = None
         self._schema: Optional[Schema] = None
+        self._external = {}
+
+        for tool in load_tools:
+            tool_name, _, tool_uri = tool.partition("=")
```

Comment:
How about we pass `load_tools` as a `dict[str,str]` or `dict[str,RemotePlugin]` here instead of using `str.partition`? Probably calling `str.partition` is more related to the CLI layer...

What do you think?